pisces/models.py

```diff
@@ -25,42 +27,123 @@
 from sklearn.model_selection import LeaveOneOut
 from sklearn.preprocessing import StandardScaler
 from concurrent.futures import ProcessPoolExecutor
+from sklearn.ensemble import RandomForestClassifier
 from .data_sets import DataSetObject, ModelInput1D, ModelInputSpectrogram, ModelOutputType, DataProcessor
 
 # %% ../nbs/02_models.ipynb 6
 class SleepWakeClassifier(abc.ABC):
-    """
+    """ Abstract class for sleep/wake classifiers. 
     """
     @abc.abstractmethod
-    def get_needed_X_y(self, data_set: DataSetObject, id: str) -> Tuple[np.ndarray, np.ndarray] | None:
-        pass
-    def train(self, examples_X: List[pl.DataFrame] = [], examples_y: List[pl.DataFrame] = [], 
-              pairs_Xy: List[Tuple[pl.DataFrame, pl.DataFrame]] = [], 
-              epochs: int = 10, batch_size: int = 32):
-        pass
+    def __init__(self, model=None, data_processor=None):
+        self.model = model
+        self.scaler = StandardScaler()
+        self.pipeline = Pipeline([('scaler', self.scaler), ('model', self.model)])
+        self.data_processor = data_processor
+
+    def _input_preprocessing(self, X: np.ndarray) -> np.ndarray:
+        return self.scaler.transform(X)
+
     def predict(self, sample_X: np.ndarray | pl.DataFrame) -> np.ndarray:
-        pass
+        """
+        Assumes data is already preprocessed using `get_needed_X_y`
+        """
+        return self.model.predict(self._input_preprocessing(sample_X))
```

Comment:
how is this different from `self.pipeline.predict`? see my comments about `self.model` not existing in some subclasses of this ABC, which limites the re-usability of this function body except when there's a `model`

Reply:
I think here we should have `self.pipeline.predict` which uses the trained scaler as well.